core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalAppender.java

```diff
@@ -40,46 +39,38 @@ public class RaftJournalAppender implements Closeable {
   /** Hosting server for the appender. Used by default for appending log entries. */
   private final RaftServer mServer;
   /** local client ID, provided along with hosting server.  */
-  private ClientId mLocalClientId;
+  private final ClientId mLocalClientId;
   /** Remote raft client. */
   private final Supplier<RaftClient> mClientSupplier;
   private volatile RaftClient mClient;
-  /** Whether to use remote RaftClient for appending log entries. */
-  private boolean mEnableRemoteClient;
 
   /**
    * @param server the local raft server
    * @param clientSupplier a function for building a remote raft client
    * @param localClientId the client id for local requests
-   * @param configuration the server configuration
    */
   public RaftJournalAppender(RaftServer server, Supplier<RaftClient> clientSupplier,
-      ClientId localClientId, InstancedConfiguration configuration) {
+      ClientId localClientId) {
     mServer = server;
     mClientSupplier = clientSupplier;
     mLocalClientId = localClientId;
-    mEnableRemoteClient = configuration.getBoolean(
-        PropertyKey.MASTER_EMBEDDED_JOURNAL_WRITE_REMOTE_ENABLED);
   }
 
   /**
    * Sends a request to raft server asynchronously.
    * @param message the message to send
-   * @param timeout the time duration to wait before giving up on the request
    * @return a future of the server reply
-   * @throws IOException if an exception occured while sending the request
+   * @throws IOException if an exception occurred while sending the request
    */
-  public CompletableFuture<RaftClientReply> sendAsync(Message message,
-      TimeDuration timeout) throws IOException {
-    if (mEnableRemoteClient) {
+  public CompletableFuture<RaftClientReply> sendAsync(Message message) throws IOException {
+    if (ServerConfiguration.getBoolean(PropertyKey.MASTER_EMBEDDED_JOURNAL_WRITE_REMOTE_ENABLED)) {
```

Comment:
this configuration seems to be static, so you might consider put it into a private static variable instead.

Reply:
What is the benefit of this approach? My initial reaction is that would make the code harder to read because the `PropertyKey` would first be assigned to a variable, and then the variable would be used, rather than using the `PropertyKey` directly.

Reply:
I was sort of getting the feeling that some property keys are valid to be changed while running are some others are not. So in the case of something able to be changed while running you should call the property key getter. For example for a message timeout you may want a user to be able to increase the timeout while running if for example the user sees their network is becoming unstable. But for a something that would break the state of the system, maybe it's better to set a final variable once for when the key is read. Or maybe this is something that could be indicated by the property key itself, that it should be only set once. I don't know if there are any existing guidelines for this though.

Reply:
I'm inclined to put them to static variables because the call is not exactly cheap. Though depending on the frequency of the invocation of this function, the overhead probably can be neglected. So this is more of a nits. If you give the variable a good name, it's not necessarily harder to read, and it comes with the benefit of THIS_IS_DEFINITELY_A_CONSTANT.

Reply:
I'm not sure whether we support changing config values at run time. I didn't see code that periodically reload configs from file so I'm not sure. If the value can be changed, then this cannot be static.

Reply:
I added `Objects.requireNonNull` on the classes I've modified where appropriate and moved made `PropertyKey` accesses a one-time thing in `private static final` variables.